#larger 
common:
    lid_topic:  "/velodyne_points"
    # cmu_rc7/imu/data   --> Laurel_Caverns
    # /imu/data   --> Long_Corridor
    # cmu_sp1/imu/data   --> MultiFloor
    imu_topic:  "/imu/data"
    time_sync_en: false         # ONLY turn on when external time synchronization is really not possible
    icp_en: true

preprocess:
    lidar_type: 2                # 1 for Livox serials LiDAR, 2 for Velodyne LiDAR, 3 for ouster LiDAR,
    scan_line: 16
    scan_rate: 10                # only need to be set for velodyne, unit: Hz,
    blind: 0.8
    point_filter_num: 1
 
mapping:
    down_sample_size: 0.2
    max_iteration: 4
#    voxel_size: 1.0
#    max_layer: 4                # 4 layer, 0, 1, 2, 3
    voxel_size: 0.25
    max_layer: 2                # 4 layer, 0, 1, 2, 3
    layer_point_size: [ 5, 5, 5, 5, 5 ]
    plannar_threshold: 0.01
    max_points_size: 1000
    max_cov_points_size: 1000
    init_gravity_with_pose: true
# 0.999212900, -0.000519121,  0.004000000, 0.080000000,
#          0.000516111,  0.999218492, -0.000939132, 0.029000000,
#         -0.004000000,  0.000802565,  0.999993652, 0.030000000,
#          0.000000000,  0.000000000,  0.000000000, 1.000000000
    fov_degree:    360
    det_range:     100.0
    extrinsic_est_en:  true      # true: enable the online estimation of IMU-LiDAR extrinsic,
    extrinsic_T: [ 0.0, 0.0, 0.0 ]
    extrinsic_R: [ 1, 0, 0,
                   0, 1, 0,
                   0, 0, 1 ]
    merge_mode: true               
    merge_distance_threshold: 0.04
    merge_bias_threshold: 1.0
    merge_layers: 10            


noise_model:
# 如果观测数据质量高（传感器精度高），降低R的相关参数值（例如ranging_cov, angle_cov）。
# 如果预测模型的鲁棒性较强，提高Q的相关参数值（例如acc_cov, gyr_cov）。
    ranging_cov: 0.04
    angle_cov: 0.1
    # v2的硬件 imu cov调低一点效果好
#    acc_cov: 0.05
#    gyr_cov: 0.0025
    acc_cov: 0.1
    gyr_cov: 0.01 
    #    acc_cov: 1.0
    #    gyr_cov: 0.5
#    b_acc_cov: 0.0043
#    b_gyr_cov: 0.000266
    b_acc_cov: 0.0000043
    b_gyr_cov: 0.000000266


publish:
    pub_voxel_map: true
    publish_max_voxel_layer: 2         # only publish 0,1,2 layer's plane
    path_en:  true
    publish_limit_z: 3.0
    publish_dense_skip: 2
    scan_publish_en:  true       # false: close all the point cloud output
    dense_publish_en: true # false: low down the points number in a global-frame point clouds scan.
    scan_bodyframe_pub_en: true  # true: output the point cloud scans in IMU-body-frame
    intensity_th: 2.0

pcd_save:
    pcd_save_en: false
    interval: -1                 # how many LiDAR frames saved in each pcd file;
    # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too much frames.
gb_icp:
    convergence_criterion: 0.0005
    max_num_iterations: 500
    min_motion_th: 0.1
    initial_threshold: 2.0
    max_range: 100
    min_range: 0.0
    max_points_per_voxel: 100
